src/main/java/org/tptacs/domain/entities/Order.java

```diff
@@ -27,4 +28,18 @@ public void addItem(ItemOrder itemOrder){
         this.lastUpdate = LocalDateTime.now();
         items.add(itemOrder);
     }
+
+    public void removeItem(Item item){
+        this.lastUpdate = LocalDateTime.now();
+
+        Iterator<ItemOrder> iterator = items.iterator();
+        while (iterator.hasNext()) {
+            ItemOrder itemOrder = iterator.next();
+            if (itemOrder.getItem().equals(item)) {
+                // Found a matching ItemOrder, remove it from the list
+                iterator.remove();
+                return; // Item found and removed, exit the loop
+            }
+        }
+    }
```

Comment:
Todo este método se puede reemplazar por esta línea para más declaratividad:
`        this.items.removeIf(io -> io.getItem().getId().equals(item.getId()))
        this.lastUpdate = LocalDateTime.now();`

Reply:
También se puede evaluar el resultado del método removeIf, que es true si algún elemento es eliminado. En caso de que se retorne false, es porque se pasó un itemId inválido. Eso podría lanzar excepción: throw new NotFoundException(item.getId(), "item")